mdal/frmts/mdal_hdf5.hpp

```diff
@@ -132,8 +134,17 @@ class HdfGroup
   protected:
     std::vector<std::string> objects( H5G_obj_t type ) const;
 
+  private:
+    HdfGroup( HdfFile::SharedHandle file, const std::string &path );
+    HdfGroup( std::shared_ptr<Handle> handle, HdfFile::SharedHandle file );
+
+  private:
+    HdfFile::SharedHandle mFile; //must be the last destroyed
```

Comment:
I think the commend it misleading, I would remove it. If you want to add comment, I would say do not destroy from this class

Reply:
yes, the comment should be "must be declared before `std::shared_ptr<Handle> d` to be sure it will be the last "